### PR TITLE
Add `MetricGroup` feature

### LINKED ISSUE #3264: To introduce `MetricGroup` class
## 🚀 Feature

Hi there!
It might be useful to group some metrics together to have less code and more coherence. My use case was using some ignite metrics in HuggingFace Trainer API which expects a single callable for computing metrics. Having this feature, that could be easily realizable with Ignite. `MetricGroup` could be something like this:

```python
from typing import Any, Dict
from ignite.metrics import Metric


class MetricGroup(Metric):
  _state_dict_all_req_keys = ('metrics',)

  def __init__(self, metrics:Dict[str, Metric]):
    self.metrics = metrics
    super(MetricGroup, self).__init__()

  def reset(self):
    for m in self.metrics.values():
      m.reset()

  def update(self, output):
    for m in self.metrics.values():
      m.update(m._output_transform(output))

  def compute(self) -> Dict[str, Any]:
    return {k: m.compute() for k,m in self.metrics.items()}
```

Usage:

```python
metric = MetricGroup({'acc': Accuracy(), 'perplexity': Perplexity()})
metric.attach(engine) # or sth in HF Trainer API
```

### PR DESCRIPTION
**Pull request by sadra-barikbin**
- **Initial commit**
- **Add tests**

Hi there!

To add `MetricGroup` feature.

Closes #3264

### CODE REVIEW

File: ignite/metrics/metric_group.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. `output_transform` of each metric in the group is also
+            called upon its update.
+
+    Examples:
+        We construct a group of metrics, attach them to the engine at once and retrieve their result.
+
+        .. code-block:: python
+            metric_group = {'acc': Accuracy(), 'precision': Precision(), 'loss': Loss(nn.NLLLoss())}
```
Comment (vfdev-5) [inline ignite/metrics/metric_group.py:21]: You may be missing `MetricGroup` here?

**Comment by vfdev-5** [comment]
Hey @sadra-barikbin thanks for the PR! Would you mind to fix the CI as well ?

**Comment by sadra-barikbin** [comment]
Seemingly it's in rouge metric, related to Nltk.

https://github.com/nltk/nltk/pull/3290

https://github.com/nltk/nltk/issues/3266

File: ignite/metrics/metric_group.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            # Metrics individually available in `state.metrics`
+            state.metrics["acc"], state.metrics["precision"], state.metrics["loss"]
+
+            # And also altogether
+            state.metrics["eval_metrics]
+    """
+
+    _state_dict_all_req_keys = ("metrics",)
+
+    def __init__(self, metrics: Dict[str, Metric], output_transform: Callable = lambda x: x):
```
Comment (vfdev-5) [inline ignite/metrics/metric_group.py:38]: Do we use this `output_transform` arg somewhere finally ?

Comment (sadra-barikbin) [reply ignite/metrics/metric_group.py:38]: It's called by `Metric.iteration_completed()`. As a way transform the output for all the metrics in group, which each one could have its specific output transform as well.

**Review by vfdev-5 — APPROVED** [review]
Thanks @sadra-barikbin, LGTM!
